src/katib.py

```diff
@@ -0,0 +1,160 @@
+import json
+from functools import wraps
+from textwrap import dedent
+from typing import List
+
+from .data import Data
+
+
+def trial_template(name, image, replicas, command, mount_path, pr=None):
+    target = "pull/{pr}/head:{pr}".format(pr=pr) if pr else "master"
+    revision = pr if pr else "master"
+    args = ["cd data/kubernetes-analysis ;",
```

Comment:
You have to `&&` here instead of `;` to fail early.

Reply:
I recommend to combine the clone and checkout to have them in one place from a source code perspective.